app/components/post_list/post_list.js

```diff
@@ -343,18 +347,41 @@ export default class PostList extends PureComponent {
         );
     };
 
+    onContentSizeChange = (contentWidth, contentHeight) => {
+        this.contentHeight = contentHeight;
+        this.props.onContentSizeChange(this.state.postListHeight, contentHeight);
+
+        if (this.state.postListHeight && contentHeight < this.state.postListHeight && this.props.extraData) {
+            // We still have less than 1 screen of posts loaded with more to get, so load more
+            this.props.onLoadMoreUp();
+        }
+    };
+
     onLayout = (event) => {
         const {height} = event.nativeEvent.layout;
-        this.setState({
-            postListHeight: height,
-        });
+        this.setState({postListHeight: height});
+    };
+
+    onScroll = (event) => {
+        const contentOffset = event.nativeEvent.contentOffset.y;
+        if (contentOffset > 0) {
+            this.pageOffsetY = contentOffset;
+            this.contentHeight = event.nativeEvent.contentSize.height;
+            const direction = this.contentOffsetY < contentOffset ? 'up' : 'down';
```

Comment:
Can we use the constant `VISIBILITY_SCROLL_UP/DOWN` here or it's different in context?